From patchwork Tue Sep  6 19:48:46 2022
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Liam Howlett <Liam.Howlett@Oracle.com>
X-Patchwork-Id: 12968170
Return-Path: <owner-linux-mm@kvack.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from kanga.kvack.org (kanga.kvack.org [205.233.56.17])
	by smtp.lore.kernel.org (Postfix) with ESMTP id E6BE2C38145
	for <linux-mm@archiver.kernel.org>; Tue,  6 Sep 2022 20:45:38 +0000 (UTC)
Received: by kanga.kvack.org (Postfix)
	id 44C188D0009; Tue,  6 Sep 2022 16:45:38 -0400 (EDT)
Received: by kanga.kvack.org (Postfix, from userid 40)
	id 3D3C08D0002; Tue,  6 Sep 2022 16:45:38 -0400 (EDT)
X-Delivered-To: int-list-linux-mm@kvack.org
Received: by kanga.kvack.org (Postfix, from userid 63042)
	id 1D8548D0009; Tue,  6 Sep 2022 16:45:38 -0400 (EDT)
X-Delivered-To: linux-mm@kvack.org
Received: from relay.hostedemail.com (smtprelay0010.hostedemail.com
 [216.40.44.10])
	by kanga.kvack.org (Postfix) with ESMTP id 08C938D0002
	for <linux-mm@kvack.org>; Tue,  6 Sep 2022 16:45:38 -0400 (EDT)
Received: from smtpin02.hostedemail.com (a10.router.float.18 [10.200.18.1])
	by unirelay10.hostedemail.com (Postfix) with ESMTP id 924C4C091C
	for <linux-mm@kvack.org>; Tue,  6 Sep 2022 20:45:37 +0000 (UTC)
X-FDA: 79882841514.02.731E236
Received: from mx0a-00069f02.pphosted.com (mx0a-00069f02.pphosted.com
 [205.220.165.32])
	by imf13.hostedemail.com (Postfix) with ESMTP id BB21A20074
	for <linux-mm@kvack.org>; Tue,  6 Sep 2022 20:45:36 +0000 (UTC)
Received: from pps.filterd (m0246617.ppops.net [127.0.0.1])
	by mx0b-00069f02.pphosted.com (8.17.1.5/8.17.1.5) with ESMTP id
 286IdYjB029914;
	Tue, 6 Sep 2022 19:48:55 GMT
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=oracle.com;
 h=from : to : cc :
 subject : date : message-id : references : in-reply-to : content-type :
 content-transfer-encoding : mime-version; s=corp-2022-7-12;
 bh=NtONvX9JVH0tbWktqIJYRovUYQEMbjAc2UkD0SznAfo=;
 b=Z6p53a5aFKy3cspz9JZBNecu+fNl2qfqbWriZzlusQmw46Z92AXCqq53cK+bkZkWFRcP
 FgMr2w+Uvn8Uqlql5Cya6Gj814CIb9/Bb66fu5ocwF19c2qx1djZOXIi/RjS39DzFCd8
 ufqd1JqrjhW/gyOtSOaDwuQ4diFslUWvnpw6qiG4oyylRCEjXdBHBI90rQaS0H6NYK2p
 1e69plVsM9Ro69M+Jnj27Lijf5B7n9cz9pwsIBZxI5wTenYRAZGDa9zPjVxw3hG0GkXB
 1xNDPgjuWr0BR1Inj0EMNhLJVxCE9MZFiEjgnT7AhqozMs4EISUr5Qc/yLi2RNSd6zCU Ow==
Received: from iadpaimrmta01.imrmtpd1.prodappiadaev1.oraclevcn.com
 (iadpaimrmta01.appoci.oracle.com [130.35.100.223])
	by mx0b-00069f02.pphosted.com (PPS) with ESMTPS id 3jbyftpvaf-1
	(version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256 verify=OK);
	Tue, 06 Sep 2022 19:48:55 +0000
Received: from pps.filterd
 (iadpaimrmta01.imrmtpd1.prodappiadaev1.oraclevcn.com [127.0.0.1])
	by iadpaimrmta01.imrmtpd1.prodappiadaev1.oraclevcn.com (8.17.1.5/8.17.1.5)
 with ESMTP id 286H8KXL023727;
	Tue, 6 Sep 2022 19:48:53 GMT
Received: from nam12-dm6-obe.outbound.protection.outlook.com
 (mail-dm6nam12lp2174.outbound.protection.outlook.com [104.47.59.174])
	by iadpaimrmta01.imrmtpd1.prodappiadaev1.oraclevcn.com (PPS) with ESMTPS id
 3jbwc9kr9w-3
	(version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256 verify=OK);
	Tue, 06 Sep 2022 19:48:53 +0000
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=IBKTVByc77EdLqVPkw9ghYLnm5JVpdcVaYUf2S9stb3M7Ku6rJwOVj+MfGR1uUuDPknqTwBNdH7sR9r1VhDgJ/TReK1ggHjq7/DiD+FGeNWUo6e4gccI5RS9WwK9pswFFizj9M2fW0xSP4B0cUGB1ceUP+37Ve70akC+lMVPManoUw1q/9k6xnWRYJ88rfpX9bRmbXEb87Zv57nVCluDng3LvO9vUiwquvFCNI5L0v+M4QQz3CE6f8iBdDPDek4ur1RQy0SBk5CCS9JJlgFdvTNrZmuoEjny2fuHjKk+iKjVooPzjYyYw6apU+tamvkiggOtzQB4JYlcmf3bqTxHAw==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-AntiSpam-MessageData-ChunkCount:X-MS-Exchange-AntiSpam-MessageData-0:X-MS-Exchange-AntiSpam-MessageData-1;
 bh=NtONvX9JVH0tbWktqIJYRovUYQEMbjAc2UkD0SznAfo=;
 b=T/PIo1yCPRFfgfJ9rysI6LiaEQRFloYPpaMqI8nagSDUzNRKZMgkQhr0F5pLlzbmqNPKYH4S71jyOoOPJ6uJo6iS8QqZ1AbDTV8u531nY9XBr9kisuFmuWhN3ujjcw9mUVkCHlwSlB8RaRjdRYiYtxUToqhQ1Fn6ZYqIPKnD80rhmvaTUer1aoHXGVZB6EEqG75g/jGrAqh6E5li9gduQxGmehplIVESBEZh/+cPrxYKZ+fSQ9Qa1nK1hJ/1eMZYz7/qkM/0bFA6U8XOxEDskeT5JT5aeIJPS/JQTW1qqmh/ENEo5iyTGN6h/PtbnL9bsYU+SIB+0LcSYsbril86Fw==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass
 smtp.mailfrom=oracle.com; dmarc=pass action=none header.from=oracle.com;
 dkim=pass header.d=oracle.com; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=oracle.onmicrosoft.com; s=selector2-oracle-onmicrosoft-com;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=NtONvX9JVH0tbWktqIJYRovUYQEMbjAc2UkD0SznAfo=;
 b=tGoMM99b34PDzkG5rm15Qcq51YfHscjvZ2GfcukUFaoEmJ6VIBFY7gcFMhF4V6d2akNrUdYa1aLyihLD5VZNnrd7xgL0qTwfQ0UU7vySUjAU2iIq48yK+oA9sLcVPlWDDbctmJ76PbapyFelgwJRYPHfHqR1jtMlW2vPxekSIe8=
Received: from SN6PR10MB3022.namprd10.prod.outlook.com (2603:10b6:805:d8::25)
 by DM4PR10MB6183.namprd10.prod.outlook.com (2603:10b6:8:8b::22) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.5588.10; Tue, 6 Sep
 2022 19:48:51 +0000
Received: from SN6PR10MB3022.namprd10.prod.outlook.com
 ([fe80::a420:3107:436d:d223]) by SN6PR10MB3022.namprd10.prod.outlook.com
 ([fe80::a420:3107:436d:d223%5]) with mapi id 15.20.5588.018; Tue, 6 Sep 2022
 19:48:51 +0000
From: Liam Howlett <liam.howlett@oracle.com>
To: "maple-tree@lists.infradead.org" <maple-tree@lists.infradead.org>,
        "linux-mm@kvack.org" <linux-mm@kvack.org>,
        "linux-kernel@vger.kernel.org"
	<linux-kernel@vger.kernel.org>,
        Andrew Morton <akpm@linux-foundation.org>
CC: "Matthew Wilcox (Oracle)" <willy@infradead.org>,
        Liam Howlett
	<liam.howlett@oracle.com>,
        Vlastimil Babka <vbabka@suse.cz>,
        David
 Hildenbrand <david@redhat.com>,
        Davidlohr Bueso <dave@stgolabs.net>
Subject: [PATCH v14 10/70] mmap: use the VMA iterator in
 count_vma_pages_range()
Thread-Topic: [PATCH v14 10/70] mmap: use the VMA iterator in
 count_vma_pages_range()
Thread-Index: AQHYwimtCpoDIcdKFU2YMeuUYYkSNg==
Date: Tue, 6 Sep 2022 19:48:46 +0000
Message-ID: <20220906194824.2110408-11-Liam.Howlett@oracle.com>
References: <20220906194824.2110408-1-Liam.Howlett@oracle.com>
In-Reply-To: <20220906194824.2110408-1-Liam.Howlett@oracle.com>
Accept-Language: en-US
Content-Language: en-US
X-MS-Has-Attach: 
X-MS-TNEF-Correlator: 
x-mailer: git-send-email 2.35.1
x-ms-publictraffictype: Email
x-ms-office365-filtering-correlation-id: 05b0e375-3ed6-4e14-0181-08da9040d2a8
x-ms-traffictypediagnostic: DM4PR10MB6183:EE_
x-ms-exchange-senderadcheck: 1
x-ms-exchange-antispam-relay: 0
x-microsoft-antispam: BCL:0;
x-microsoft-antispam-message-info: 
 g8Wu54bt4IoXogFpne+iANUaIB2A3ybj5UC1TIDLX3WAOTk1g/F2ydbilEi0qUcynOb75Q8o5Iog9MOMwf2OuxguzLh3reQiq3AJ+SQcIlA2jnZ3VfQ3YK1srm1R5bQN+NE1OrRET6+b29OeBCWZeTlAklf8d6GssOYqk6WtvRnr3Na5M8xE8F/ZgWW06W9MRzlImUt5ECH6ReQydk8zTgs6gGPf5eK8q59X7OJal9fdu91BCkujyn/jL4hGF/z2X7nfI5N/n3r5Sp9ChXdtw2yKyYxqzUN/LCsuciN+dPktPen/GZ4DwkrpxHmQ064UATF0WzgaVUbqUYziD/MiNku3Y9GYaxp3CCh5g/OtQXJyRPJMRz1f4FQ37K7xcPeQfFL8Gn4NQ+E7jNnJs7nqlE4FDvA0gz0KYRovwonN+4doZ02SLh577TnBs4dfWY876qIbsoYpFnvHkE2tBl6RpBMsqPSZF344N2BLgHTPdtqd6km9yQ+y2fUXgVFQHRkzgkeyiAPkLPmpFpc6AVxyy7ni+QVOF3Fk5Jwvx2ZQydKMtGvIwgZV5XtIqRonGOtHCNix+b9uGnYotZBV6BNVV1iRH3PxMqdOMZ7Rtz2QqJKUZvuykgh+lZ/6hnH367GEGvoDo0lfYWwN9HycOmNxTXGKZnq/XyCsl3w6WKByY2Kr1s31YoiYv8KzXLW7k4GrUKperPd2uktuwESmNLEb4lXO8D9ZJD55dm6C+amcWBnDWtJ3Sa7noOGlWb8hS0RVHx6C+FCo98iTkY2FazZIgw==
x-forefront-antispam-report: 
 CIP:255.255.255.255;CTRY:;LANG:en;SCL:1;SRV:;IPV:NLI;SFV:NSPM;H:SN6PR10MB3022.namprd10.prod.outlook.com;PTR:;CAT:NONE;SFS:(13230016)(39860400002)(366004)(376002)(396003)(346002)(136003)(66556008)(38100700002)(122000001)(8936002)(5660300002)(76116006)(66946007)(66446008)(66476007)(2906002)(83380400001)(8676002)(91956017)(36756003)(2616005)(186003)(86362001)(1076003)(110136005)(44832011)(71200400001)(6666004)(38070700005)(54906003)(26005)(64756008)(41300700001)(6506007)(6512007)(6486002)(316002)(478600001)(4326008);DIR:OUT;SFP:1101;
x-ms-exchange-antispam-messagedata-chunkcount: 1
x-ms-exchange-antispam-messagedata-0: =?iso-8859-1?q?8Cm1Hd/h86szRNba4Gss/G+?=
	=?iso-8859-1?q?yDCHza/AwIqnLH8Ly/rBpDxRIWXB65C8coPD7OuYNAPKXuiaD0GH7uJsdrif?=
	=?iso-8859-1?q?QVOsABLBMaXKTimFdSshaieSmC3sDKzw7B8tQeULFQSYYkoeJaLgMNMhYe/U?=
	=?iso-8859-1?q?mcQDzAY+1bNHTnStiOEj4tJbVESBwOdPYp16QCXqpsvvtvRKpxQm3QNW190r?=
	=?iso-8859-1?q?hF5VJr81wT3kZAyoV6g7Eec/Nn9Y3dWzEKSy7Uc10g4UcxLsWcujRaflYdL4?=
	=?iso-8859-1?q?y+tY7sc0l5MBb9sSLJGBaWsttsbEkLfnX1I3L2MQvbY4q2C6D3NSLrdjTUaF?=
	=?iso-8859-1?q?JzbE5ckyzqTUuh/b5pA6mKc+rJITOncLdueJczaRaM+C5NxI+UApjj6P6JNw?=
	=?iso-8859-1?q?bKuUXpmCaWJTaDOLTx9O2mk3QINg04tls76N2YxwQwD7ULMPwv4x7VH0tcCK?=
	=?iso-8859-1?q?+srOjpeTebctK1bt57cTZtIp+5XX3at/PF1pxKwBiMaQ0x12sQtutU/Z9knW?=
	=?iso-8859-1?q?qRL7A1NK3LqhF7z+yrKlQdWYurnYc0drwaWbHjx+JDE9x5Ga+wv4nNIx4HqD?=
	=?iso-8859-1?q?V9Lh1uLhIhawIIJxMzlgCzpfiJtVpiAYEzB+3+uQSYx/PLy0SCqlFP2NP7XW?=
	=?iso-8859-1?q?AfDBdMkT2YgEeK1QllxJ0FhVyoIgMG5k9jHqcNvHbB/CM0+uBL6zvxTWmM1k?=
	=?iso-8859-1?q?KToPP72cfteNv7T9ZT/0xcVe3BlJD85MeLzw7rWrtzbLHOIZrlRLYeBPX4rl?=
	=?iso-8859-1?q?3/ELJv0iNZhFHK27twzYjwG81M2ygj0QRVhlZDkOPuOrNgs1TBRtohpaVNg9?=
	=?iso-8859-1?q?TxjEuDzncQWkJKdouKJTsAwhUMQTRaz8LYL+E+TETAxzyTAPBOFyOF4i/M06?=
	=?iso-8859-1?q?RhPb9UPJ9PGmqQa9Fzo2ZP2AUslNXMZ2oWcMaBMsxlXlXRjhLLepfchrCStF?=
	=?iso-8859-1?q?fvPE6h7C4k3xwnB0UQJ2rT/lj54q/1EmpoZYbA64e3ImdZUq5GM6NzN9j3jC?=
	=?iso-8859-1?q?FgbC17+CVYh3Q3L0T+HRgcuVDGTK0H8gzhS2FJaW4ALPncpdPlJS5hEBz/Mz?=
	=?iso-8859-1?q?MsRIHGe6D7mPTRefqprNndpiqpwGt9ZhHinUhp6Ru+NTN9/dzZrUdRUKOVZJ?=
	=?iso-8859-1?q?ayCXbmUnf390ZCgpRMQkOywaTFAMjFj7YA1G2AvJGEu+v+EwwPyq9Q1ii5rO?=
	=?iso-8859-1?q?buPfgTJX/iaP2WUZZzBEv0i2IT8Dz/FiWdRLEhCf8mzfKPcTqOpmynKuAf/H?=
	=?iso-8859-1?q?s0/q1FdyXfApu5B7slAfIYtSqGwkHiKhSEB+VQzPsKFFfqxkpeRGzjfZl361?=
	=?iso-8859-1?q?2x7fF7TK5nhABMJ/XFFvylfz8WxnA4rNQDLTkqfWNdtWdPSX6eBkRo+AjNvv?=
	=?iso-8859-1?q?fS6sjvz6vTeSYWNsFHt40eRruFPDHtAsckTKwwbT2VUnIg9y1oU3eJwPkBZ7?=
	=?iso-8859-1?q?F4Zjnsx50ozDtFN6kSbo/9lSb0rV5FVdLRssVUf1pv/aSVtWC/8rqpTgRjJH?=
	=?iso-8859-1?q?/cvNyQ3VtDfNucYRemmid6Q+mdZ8ep6ygQqd/bIpXyjLNOdx1wZAmlarzmjn?=
	=?iso-8859-1?q?QqHXuT1crCAtbAxy9zWCwdAzVDVoxPx0ZYiROostD+AzM/IrxAT0+7sPZCtu?=
	=?iso-8859-1?q?YDMP978qUoXIpRM87S/GubEhMMizetQ83nvZ+GQ=3D=3D?=
MIME-Version: 1.0
X-OriginatorOrg: oracle.com
X-MS-Exchange-CrossTenant-AuthAs: Internal
X-MS-Exchange-CrossTenant-AuthSource: SN6PR10MB3022.namprd10.prod.outlook.com
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 05b0e375-3ed6-4e14-0181-08da9040d2a8
X-MS-Exchange-CrossTenant-originalarrivaltime: 06 Sep 2022 19:48:46.4913
 (UTC)
X-MS-Exchange-CrossTenant-fromentityheader: Hosted
X-MS-Exchange-CrossTenant-id: 4e2c6054-71cb-48f1-bd6c-3a9705aca71b
X-MS-Exchange-CrossTenant-mailboxtype: HOSTED
X-MS-Exchange-CrossTenant-userprincipalname: 
 s5pMbOxSw+z0gFpAdWkWjn5B4OlsVk4H8S6L9jitBGneAokHSSB0Sq+LHHF/HjvwSGx4BDZ8eduB3BX/ho44cA==
X-MS-Exchange-Transport-CrossTenantHeadersStamped: DM4PR10MB6183
X-Proofpoint-Virus-Version: vendor=baseguard
 engine=ICAP:2.0.205,Aquarius:18.0.895,Hydra:6.0.528,FMLib:17.11.122.1
 definitions=2022-09-06_09,2022-09-06_02,2022-06-22_01
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0 phishscore=0
 mlxscore=0 adultscore=0
 malwarescore=0 mlxlogscore=999 bulkscore=0 suspectscore=0 spamscore=0
 classifier=spam adjust=0 reason=mlx scancount=1 engine=8.12.0-2207270000
 definitions=main-2209060091
X-Proofpoint-ORIG-GUID: Bxv-fSoYTuhni7GCb2VJXI_JBh9SQ59f
X-Proofpoint-GUID: Bxv-fSoYTuhni7GCb2VJXI_JBh9SQ59f
ARC-Authentication-Results: i=2;
	imf13.hostedemail.com;
	dkim=pass header.d=oracle.com header.s=corp-2022-7-12 header.b=Z6p53a5a;
	dkim=pass header.d=oracle.onmicrosoft.com
 header.s=selector2-oracle-onmicrosoft-com header.b=tGoMM99b;
	arc=pass ("microsoft.com:s=arcselector9901:i=1");
	spf=pass (imf13.hostedemail.com: domain of liam.howlett@oracle.com designates
 205.220.165.32 as permitted sender) smtp.mailfrom=liam.howlett@oracle.com;
	dmarc=pass (policy=none) header.from=oracle.com
ARC-Seal: i=2; s=arc-20220608; d=hostedemail.com; t=1662497136; a=rsa-sha256;
	cv=pass;
	b=H5Vaq0EG4ol0UoiC8jeqm7Yt6LoUPECTUFbjZ/AHOgfqvCn8pJYAI/Zv06e6lmuBgRBhG+
	BG4ZBSof7CTlyhVKAtLTrflMdkys0ePBumJ/JZfIIn0dY/UrixzSt+QGc8zserD23ZbOEZ
	VNuYjRh9HFovm14wO+e/BoBgonqHeJA=
ARC-Message-Signature: i=2; a=rsa-sha256; c=relaxed/relaxed;
 d=hostedemail.com;
	s=arc-20220608; t=1662497136;
	h=from:from:sender:reply-to:subject:subject:date:date:
	 message-id:message-id:to:to:cc:cc:mime-version:mime-version:
	 content-type:content-type:
	 content-transfer-encoding:content-transfer-encoding:
	 in-reply-to:in-reply-to:references:references:dkim-signature;
	bh=NtONvX9JVH0tbWktqIJYRovUYQEMbjAc2UkD0SznAfo=;
	b=vL+HuPnDOqJBXymR2AYMO7HuOgifn+kgrBbglqx9DiMP29mypfHdwfBEC4Hkvl9CPuFRKd
	/fDMcKEtgZ3VYKsu9JKrHHJs3kSk4LKZfdk9zjiyq9petmqQsb+D4aaVpMJqX0Q+khccZ2
	q1SAa6snzutIaMy/fiLCAKOPpUgDsic=
Authentication-Results: imf13.hostedemail.com;
	dkim=pass header.d=oracle.com header.s=corp-2022-7-12 header.b=Z6p53a5a;
	dkim=pass header.d=oracle.onmicrosoft.com
 header.s=selector2-oracle-onmicrosoft-com header.b=tGoMM99b;
	arc=pass ("microsoft.com:s=arcselector9901:i=1");
	spf=pass (imf13.hostedemail.com: domain of liam.howlett@oracle.com designates
 205.220.165.32 as permitted sender) smtp.mailfrom=liam.howlett@oracle.com;
	dmarc=pass (policy=none) header.from=oracle.com
X-Stat-Signature: 7oys4odmqtcqz4f8phmh6hiuq5aa3we6
X-Rspamd-Queue-Id: BB21A20074
X-Rspamd-Server: rspam09
X-Rspam-User: 
X-HE-Tag: 1662497136-298705
X-Bogosity: Ham, tests=bogofilter, spamicity=0.000000, version=1.2.4
Sender: owner-linux-mm@kvack.org
Precedence: bulk
X-Loop: owner-majordomo@kvack.org
List-ID: <linux-mm.kvack.org>

From: "Matthew Wilcox (Oracle)" <willy@infradead.org>

This simplifies the implementation and is faster than using the linked
list.

Signed-off-by: Matthew Wilcox (Oracle) <willy@infradead.org>
Signed-off-by: Liam R. Howlett <Liam.Howlett@Oracle.com>
Acked-by: Vlastimil Babka <vbabka@suse.cz>
Reviewed-by: David Hildenbrand <david@redhat.com>
Reviewed-by: Davidlohr Bueso <dave@stgolabs.net>
---
 mm/mmap.c | 24 +++++++-----------------
 1 file changed, 7 insertions(+), 17 deletions(-)

diff --git a/mm/mmap.c b/mm/mmap.c
index 20718645d82f..f1b07751a1e4 100644
--- a/mm/mmap.c
+++ b/mm/mmap.c
@@ -629,29 +629,19 @@ munmap_vma_range(struct mm_struct *mm, unsigned long start, unsigned long len,
 
 	return 0;
 }
+
 static unsigned long count_vma_pages_range(struct mm_struct *mm,
 		unsigned long addr, unsigned long end)
 {
-	unsigned long nr_pages = 0;
+	VMA_ITERATOR(vmi, mm, addr);
 	struct vm_area_struct *vma;
+	unsigned long nr_pages = 0;
 
-	/* Find first overlapping mapping */
-	vma = find_vma_intersection(mm, addr, end);
-	if (!vma)
-		return 0;
-
-	nr_pages = (min(end, vma->vm_end) -
-		max(addr, vma->vm_start)) >> PAGE_SHIFT;
-
-	/* Iterate over the rest of the overlaps */
-	for (vma = vma->vm_next; vma; vma = vma->vm_next) {
-		unsigned long overlap_len;
-
-		if (vma->vm_start > end)
-			break;
+	for_each_vma_range(vmi, vma, end) {
+		unsigned long vm_start = max(addr, vma->vm_start);
+		unsigned long vm_end = min(end, vma->vm_end);
 
-		overlap_len = min(end, vma->vm_end) - vma->vm_start;
-		nr_pages += overlap_len >> PAGE_SHIFT;
+		nr_pages += PHYS_PFN(vm_end - vm_start);
 	}
 
 	return nr_pages;

